docs/src/design/proposals/APIRedesign.md

```diff
@@ -0,0 +1,1050 @@
+  awkward APIs or extra factory interfaces.
+
+## Remove capability delegation
+
+The `delegation` capability has only been useful for the
+`telemetry` capability, but using both together creates
+some complexity to manage. In practice, these means that
+telemetry views need to go through `telemetryHandler` to
+get their telemetry, which in turn has an awkward API.
+
+This could be resolved by:
+
+* Removing `delegation` as a capability altogether.
+* Reworking `telemetry` capability API to account for
+  the possibility of multiple telemetry-providing
+  domain objects. (Perhaps just stick `domainObject`
+  in as a field in each property of `TelemetryMetadata`?)
+* Move the behavior currently found in `telemetryHandler`
+  into the `telemetry` capability itself (either the
+  generic version, or a version specific to telemetry
+  panels - probably want some distinct functionality
+  for each.)
+
+### Benefits
+
+* Reduces number of interfaces.
+* Accounting for the possibility of multiple telemetry objects
+  in the `telemetry` capability API means that views using
+  this will be more immediately aware of this as a possibility.
+
+### Detriments
+
+* Increases complexity of `telemetry` capability's interface
+  (although this could probably be minimized.)
+
+## Nomenclature Change
+
+Instead of presenting Open MCT Web as a "framework" or
+"platform", present it as an "extensible application."
+
+This is mostly a change for the developer guide. A
+"framework" and a "platform" layer would still be useful
+architecturally, but the plugin developer's mental model
+for this would then be inclined toward looking at defined
+extension points. The underlying extension mechanism could
+still be exposed to retain the overall expressive power of
+the application.
+
+This may subtly influence other design decisions in order
+to match the "extensible application" identity. On a certain
+level, this contradicts the proposal to
+[rename views to applications](#rename-views-to-applications).
+
+### Benefits
+
+* May avoid incurring some of the "framework aversion" that
+  is common among JavaScript developers.
+* More accurately describes the application.
+
+### Detriments
+
+* May also be a deterrent to developers who prefer the more
+  "green field" feel of developing applications on a useful
+  platform.
+
+## Capabilities as Mixins
+
+Change the behavior of capabilities such that they act as
+mixins, adding additional methods to domain objects.
+Checking if a domain object has a `persistence` capability
+would instead be reduced to checking if it has a `persist`
+method.
+
+Mixins would be applied in priority order and filtered for
+applicability by policy.
+
+### Benefits
+
+* Replaces "capabilities" (which, as a concept, can be hard
+  to grasp) with a more familiar "mixins" concept, which has
+  been used more generally across many languages.
+* Reduces interface depth.
+
+### Detriments
+
+* Requires checking for the interface exposed by a domain
```

Comment:
Agree this is very similar to the current state of things (`hasCapability` is slightly different insofar as you find out about several methods instead of just one, but that's a small difference); I consider the requirement to check for capabilities a detriment of the current API as well.